aten/src/ATen/native/mkldnn/BinaryOps.cpp

```diff
@@ -46,6 +47,35 @@ Tensor& mkldnn_mul_(Tensor& self, const Tensor& other) {
 namespace at {
 namespace native {
 
+Tensor emptyBinaryOp(const Tensor& self, const Tensor& other) {
+  if (!self.requires_grad() && !other.requires_grad()) {
+    auto out_size = infer_size(self.sizes(), other.sizes());
+    auto out_dtype = promoteTypes(
```

Comment:
is it always possible to promote?

Reply:
yes. promoteTypes always succeeds. MKLDNN only supports bfloat16 & float as well.